pogs/extract.go

```diff
@@ -376,6 +397,15 @@ func isTypeMatch(r reflect.Type, s schema.Type) bool {
 	case schema.Type_Which_list:
 		e, _ := s.List().ElementType()
 		return r.Kind() == reflect.Slice && isTypeMatch(r.Elem(), e)
+	case schema.Type_Which_interface:
+		if !r.Implements(clientType) && r.Kind() == reflect.Struct {
+			field, ok := r.FieldByName("Client")
```

Comment:
I'd like for this to check for a more strict bound:

1. Has one field
2. That field is called `Client`
3. That field is of type `capnp.Client`